src/docker/restorableappengine.cc

```diff
@@ -754,4 +790,32 @@ std::string RestorableAppEngine::getContentHash(const boost::filesystem::path& p
   return boost::algorithm::to_lower_copy(boost::algorithm::hex(Crypto::sha256digest(content)));
 }
 
+bool RestorableAppEngine::areDockerAndSkopeoOnTheSameVolume(const boost::filesystem::path& skopeo_path,
+                                                            const boost::filesystem::path& docker_path) {
+  const auto skopeoVolumeID{getPathVolumeID(skopeo_path.parent_path())};
+  if (!std::get<1>(skopeoVolumeID)) {
+    LOG_WARNING << "Failed to obtain an ID of a skopeo store volume; path: " << skopeo_path
+                << ", err: " << strerror(errno);
+  }
+  const auto dockerVolumeID{getPathVolumeID(docker_path.parent_path())};
+  if (!std::get<1>(dockerVolumeID)) {
+    LOG_WARNING << "Failed to obtain an ID of a docker store volume; path: " << docker_path
+                << ", err: " << strerror(errno);
+  }
+  // we assume that a docker data root and a skopeo store are located on the same volume in case of a failure to obtain
```

Comment:
would it be safer to just `return false` right after the LOG_WARNING's above? then you aren't relying on the failed return value to behave a certain way.

Reply:
By default we assume that the stores are located on the same volume. If it fails for both we need it to be `true`. If for one of it it should be `false`.

Reply:
ah - i see the `getPathVolumeID` controls this.